contracts/xasset-orchestrator/src/orchestrator.rs

```diff
@@ -0,0 +1,179 @@
+use crate::{error::Error, Contract};
+use loam_sdk::{
+    import_contract, loamstorage,
+    soroban_sdk::{
+        self, env, xdr::ToXdr, Address, Bytes, BytesN, Env, InstanceItem, Lazy, LoamKey,
+        PersistentMap, String, Symbol,
+    },
+    subcontract,
+};
+use loam_subcontract_core::Core;
+
+import_contract!(xasset);
+
+#[loamstorage]
+pub struct Storage {
+    /// Wasm hash of the xasset contract
+    wasm_hash: InstanceItem<loam_sdk::soroban_sdk::BytesN<32>>,
+    /// XLM SAC contract address; initialized and then passed
+    /// to deployed xasset contracts
+    xlm_sac: InstanceItem<loam_sdk::soroban_sdk::Address>,
+    /// XLM oracle contract, initialized and then passed
+    /// to deployed xasset contracts
+    xlm_contract: InstanceItem<loam_sdk::soroban_sdk::Address>,
+    /// A map of deployed asset contracts to their asset symbol.
+    /// This is used to check if a contract is a valid asset contract
+    /// and to get the asset symbol from the contract address.
+    /// The key is the asset symbol, the value is the asset contract address.
+    assets: PersistentMap<loam_sdk::soroban_sdk::String, loam_sdk::soroban_sdk::Address>,
```

Comment:
One downside to our current approach is that you can't iterate over the map since it just stores the info directly in storage. So you could use a normal map here if we expect the number of assets to remain relatively small 100s. This will let you iterate over the map with a view call or just fetch it directly. It will get more expensive to read and write as it grows.

Reply:
But a view method could likely return the whole map if it's small enough.

Reply:
When you say "a normal map" -- from what crate? What is a "normal map"? `core::iter::Map`? `loam_sdk::soroban_sdk::Map`?

Reply:
Good point! I soroban map! It is stored outside of the contract but as one chunk so you have to pay to load all of it even if you only wanted to read/write one entry. But it can paginated or converted directly to a vector of tuples.